inductiva/molecules/scenarios/protein_solvation/protein_solvation.py

```diff
@@ -97,80 +72,23 @@ def gen_config(self, simulator: Simulator):
         )
 
 
-@ProteinSolvation.run_simulation.register
-def _(self, simulator: GROMACS):
-    """Run the simulation using GROMACS."""
-    #Solvation
-    simulator.run(input_dir=self.working_dir,
-                  output_directory=self.working_dir,
-                  method_name="pdb2gmx",
-                  f=self.protein_pdb,
-                  o="protein.gro",
-                  water="tip3p",
-                  user_input="6")
-    simulator.run(input_dir=self.working_dir,
-                  output_directory=self.working_dir,
-                  method_name="editconf",
-                  f="protein.gro",
-                  o="protein_box.gro",
-                  c="yes",
-                  d="1.0",
-                  bt="cubic")
-    simulator.run(input_dir=self.working_dir,
-                  output_directory=self.working_dir,
-                  method_name="solvate",
-                  cp="protein_box.gro",
-                  o="protein_solv.gro",
-                  p="topol.top")
-    simulator.run(input_dir=self.working_dir,
-                  output_directory=self.working_dir,
-                  method_name="grompp",
-                  f="ions.mdp",
-                  c="protein_solv.gro",
-                  p="topol.top",
-                  o="ions.tpr")
-    simulator.run(input_dir=self.working_dir,
-                  output_directory=self.working_dir,
-                  method_name="genion",
-                  s="ions.tpr",
-                  o="protein_solv_ions.gro",
-                  p="topol.top",
-                  pname="NA",
-                  nname="CL",
-                  neutral="yes")
-    # Energy minimization
-    simulator.run(input_dir=self.working_dir,
-                  output_directory=self.working_dir,
-                  method_name="grompp",
-                  f="energy_minimization.mdp",
-                  c="protein_solv_ions.gro",
-                  p="topol.top",
-                  o="em.tpr")
-    simulator.run(input_dir=self.working_dir,
-                  output_directory=self.working_dir,
-                  method_name="mdrun",
-                  deffnm="em",
-                  v="yes")
-    # Simulation
-    simulator.run(input_dir=self.working_dir,
-                  output_directory=self.working_dir,
-                  method_name="grompp",
-                  f="simulation.mdp",
-                  c="em.gro",
-                  r="em.gro",
-                  p="topol.top",
-                  o="solvated_protein.tpr")
-    simulator.run(input_dir=self.working_dir,
-                  output_directory=self.working_dir,
-                  method_name="mdrun",
-                  deffnm="solvated_protein",
-                  v="yes")
+@ProteinSolvation.gen_aux_files.register
+def _(self, simulator: GROMACS, input_dir, protein_pdb):  # pylint: disable=unused-argument
+    """Setup the working directory for the simulation."""
+    # Copy protein pdb to working_dir
```

Comment:
I think these comments are unnecessary, it is clear what you are doing. What you can do is explain the reason to establish a unique name to the `protein.pdb`.